src/merge_bot/merge_bot.py

```diff
@@ -414,6 +441,9 @@ def run(
 
         if update_go_modules:
             commit_go_mod_updates(gitwd)
+
+        run_make(gitwd)
```

Comment:
Shouldn't we more this under the `if update_go_modules:` conditional just above? Also it feels like we want `run_make()` to run _before_ `commit_go_mod_updates()`

Perhaps this would be good to split the functions, and have them only generate updated content, then in a separate function commit the new content if there is anything to commit.

Reply:
will do

Reply:
Then make a new option. I believe the intention is that by default we don't add more commits that the one we merge from the upstream repository. @mdbooth @pierreprinetti, any thoughts?

Reply:
I'm happy with a new function. I'll wait for Pierre & Matt in case they have any feedback otherwise I'll propose it by end of week.

Reply:
@pierreprinetti and @mdbooth  ^

Reply:
I've done the change Martin.